runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformResult.java

```diff
@@ -33,7 +33,7 @@
 /**
  * The result of evaluating an {@link AppliedPTransform} with a {@link TransformEvaluator}.
  */
-public interface TransformResult {
+public interface TransformResult<InputT, OutputT> {
```

Comment:
Done. Eliminated the `OutputT`. Kind of disappointing that `ParDo` ruins the party, since it could catch a lot of errors. I think it is worth considering ways to restore typing.

I've left it as a fixup commit for PR purposes, but the commit message will need revision.